api/apis/service_binding_handler.go

```diff
@@ -93,6 +96,49 @@ func (h *ServiceBindingHandler) createHandler(authInfo authorization.Info, w htt
 	writeResponse(w, http.StatusCreated, presenter.ForServiceBinding(serviceBinding, h.serverURL))
 }
 
+func (h *ServiceBindingHandler) listHandler(authInfo authorization.Info, w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+	w.Header().Set("Content-Type", "application/json")
+
+	if err := r.ParseForm(); err != nil {
+		h.logger.Error(err, "Unable to parse request query parameters")
+		writeUnprocessableEntityError(w, "unable to parse query")
+		return
+	}
+
+	listFilter := new(payloads.ServiceBindingList)
+	err := schema.NewDecoder().Decode(listFilter, r.Form)
+	if err != nil {
+		switch err.(type) {
```

Comment:
How about extracting the check for UnknownKeyError into a utility function so that the code below reads like

```
if isUnknownKeyError(err) {
   // return unknown key error
}
// return unknown error

Reply:
Added https://github.com/cloudfoundry/cf-k8s-controllers/issues/698